### DISCUSSION TITLE
Question

### DISCUSSION THREAD

**Discussion by MN-Dheeraj** [apache/camel-karavan#1234] [Q&A] [answered]
### Ask a question

I am building my spring boot project on karavan. I started by adding 'from' route, 'log' and then 'processsor'.
I added a custom processor that takes the class of name 'Esbloginterceptorprocessor '. I added that in my route on karavan.
The following is the yaml version of the routes built.

- route:
    id: route-a27c
    nodePrefixId: route-06e
    from:
      id: from-1df0
      uri: rest
      parameters:
        method: post
        path: /LoanAccountDetails
      steps:
        - process:
            id: process-d06b
            description: Interception and logging to DB
            ref: Esbloginterceptorprocessor

Here is the class 'Esbloginterceptorprocessor'.

import org.apache.camel.Configuration ;
import org.apache.camel.BindToRegistry;
import org.apache.camel.Exchange;
import org.apache.camel.Processor;
import org.springframework.core.env.Environment;

@Configuration 
@BindToRegistry("Esbloginterceptorprocessor ")
public class Esbloginterceptorprocessor implements Processor {
	private final Logger logger = LogManager.getLogger(getClass());
	@Autowired
	Environment env;
	public Esbloginterceptorprocessor(Environment environment) {
		env = environment;
	}
}

I also added @Configuration and @BindToRegistry to my processor class for it be binded to registry. However, Here is the error when starting the route.

_org.apache.camel.FailedToCreateRouteException: Failed to create route route-a27c at: >>> process[ref:Esbloginterceptorprocessor] <<< in route: Route(route-a27c)[From[rest:post:/LoanAccountDetails] -> [pr... because of No bean could be found in the registry for: Esbloginterceptorprocessor of type: org.apache.camel.Processor
	at org.apache.camel.reifier.RouteReifier.doCreateRoute(RouteReifier.java:228)
	at org.apache.camel.reifier.RouteReifier.createRoute(RouteReifier.java:79)
Caused by: org.apache.camel.NoSuchBeanException: No bean could be found in the registry for: Esbloginterceptorprocessor of type: org.apache.camel.Processor_

### Variant

Web Application

### Container Management (if applicable)

Docker

### Operating System (if applicable)

Windows

### Version

4.4.0

**Comment by davsclaus** [comment]
the question is badly formatted and not vert readable - can you format this better.
Also show the processor code what you did

**Reply by MN-Dheeraj** [reply]
Is the content better now? Please let me know if it needs to be formatted still.

**Comment by davsclaus** [comment, marked as answer]
You must have a default no-arg constructor, and Environment is some spring boot stuff that Karavan does not support, so you cant use that.

**Comment by davsclaus** [comment]
Just do

```
@BindToRegistry("Esbloginterceptorprocessor")
public class Esbloginterceptorprocessor implements Processor {

    public void process(Exchange exchange) throws Exception {
       // do your code here
    }
}
```

**Comment by davsclaus** [comment]
You can more easily do this from Karavan UI by clicking the plus sign in the field where you type the processor name, then Karavan creates an new empty class for you that works

**Reply by MN-Dheeraj** [reply]
Got it. Thank you.
I see adding an empty constructor resolves the issue with out any annotation and separate bean definition.
Also, I see problem with environment. For arguments of primitive java data types, I don't see problem when bean with constructors are defined.

May I know if there is an alternative solution to use spring stuff in a different way or maybe in different version of karavan?

**Reply by davsclaus** [reply]
No - karavan is pure camel only - no plan for anything else.

If you want to code spring boot then use camel spring boot and not karavan web.
Also its not ideal to write java code in a web browser, so karavan is only for low-coding / scripting and some simpler java code and for users whom does not care or know spring boot or any big complex java frameworks or build tools.

**Reply by MN-Dheeraj** [reply]
Got it, Thank you.
We are actually trying to build routes for the existing code and further help developers to write as less code as possible on karavan for new routes (new APIs).
May I understand what you mean by using camel spring boot and not karavan web. Is it still not on karavan or a normal local code that we usually do?
